js/main.js

```diff
@@ -3,28 +3,125 @@ const imgPreview = document.querySelector('.img-preview');
     notes: notes.value,
     entryId: data.nextEntryId
+
   };
 
   data.nextEntryId++;
   data.entries.unshift(newObject);
   imgPreview.setAttribute('src', './images/placeholder-image-square.jpg');
   form.reset();
+  renderEntry(newObject);
+  viewSwap('entries');
+
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
+});
+
+
+document.addEventListener('DOMContentLoaded', () => {
+  const previousData = localStorage.getItem('javascript-local-storage');
+
+  if (previousData !== null) {
+    data = JSON.parse(previousData);
+    data.entries.forEach((entry) => {
+      renderEntry(entry);
+    });
+    viewSwap(data.view);
+    if (data.entries.length > 0) {
+      toggleNoEntries();
+    }
+  } else {
+    viewSwap(data.view)
+  }
 });
+
+function renderEntry(entry) {
+  const entryList = document.querySelector('.entry-list');
+  const newLiContainer = document.createElement("li");
+
+  const newLi = document.createElement('li');
+  newLi.setAttribute('class', 'row entry-container');
+
+  const columnHalf = document.createElement('div');
+  columnHalf.setAttribute('class', 'column-half');
+
+  const imgPreview = document.createElement('img');
+  imgPreview.setAttribute('class', 'img-preview');
+  imgPreview.src = entry.url;
+
+  const entryInfoContainer = document.createElement('div');
+  entryInfoContainer.setAttribute('class', 'column-half entry-info-container');
+
+  const entryTitle = document.createElement('div');
+  entryTitle.setAttribute('class', 'entry-title');
+  entryTitle.textContent = entry.title;
+
+  const entryNotes = document.createElement('div');
+  entryNotes.setAttribute('class', 'entry-notes');
+  entryNotes.textContent = entry.notes;
+
+  columnHalf.appendChild(imgPreview);
+  entryInfoContainer.appendChild(entryTitle);
+  entryInfoContainer.appendChild(entryNotes);
+  newLi.appendChild(columnHalf);
+  newLi.appendChild(entryInfoContainer);
+  newLiContainer.appendChild(newLi);
+  entryList.appendChild(newLiContainer);
+}
+
+function toggleNoEntries() {
+  const noEntries = document.querySelector('.noEntries');
+
+  if (noEntries.classList.contains('hide')) {
+    noEntries.classList.remove('hidden');
+  } else {
+    noEntries.classList.add('hidden');
+  }
+}
+
+viewEntries.addEventListener('click', () => {
+  viewSwap('entries');
+});
+
+viewEntryForm.addEventListener('click', () => {
+  viewSwap('entry-form');
+})
+
+function viewSwap(elementToSwap) {
+  const entryForm = document.querySelector('[data-view="entry-form"]');
+  const entries = document.querySelector('[data-view="entries"]');
+
+  if (elementToSwap === 'entries') {
+    data.view = 'entries';
+
+    const dataModel = JSON.stringify(data);
+
+    localStorage.setItem('javascript-local-storage', dataModel);
+    entries.classList.remove('hidden');
+    entryForm.classList.add('hidden');
+
+  } else {
+    data.view = 'entry-form';
+
+    const dataModel = JSON.stringify(data);
+
+    localStorage.setItem('javascript-local-storage', dataModel);
```

Comment:
lines 121-123 aren't needed since your `data.js` will handle writing to localStorage on the `beforeunload` event